typescript/src/redemption.ts

```diff
@@ -0,0 +1,168 @@
+// @ts-ignore
+import bcoin from "bcoin"
+// @ts-ignore
+import wif from "wif"
+import { BigNumber } from "ethers"
+import {
+  RawTransaction,
+  UnspentTransactionOutput,
+  Client as BitcoinClient,
+} from "./bitcoin"
+
+/**
+ * Contains information needed to fulfill a redemption request.
+ */
+export interface RedemptionRequest {
+  /**
+   * The address that is the recipient of the redeemed Bitcoins.
+   * The type of the address must be P2PKH, P2WPKH, P2SH or P2WSH.
+   */
+  address: string
+
+  /**
+   * The amount of Bitcoins in satoshis that is requested to be redeemed.
+   * The actual value of the output in the Bitcoin transaction will be decreased
+   * by the sum of the fee share and the treasury fee for this particular output.
+   */
+  amount: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount in
+   * redemption request and used to pay for the transaction.
+   * The value should not be greater than the max fee in the Bridge on-chain
+   * contract at the time the redemption request was made.
+   * The sum of all output fee values makes the total fee of the redemption
+   * transaction.
+   */
+  feeShare: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount in
+   * redemption request and used to pay the treasury fee.
+   * The value should be equal exactly the value of treasury fee in the Bridge
+   * on-chain contract at the time the redemption request was made.
+   */
+  treasuryFee: BigNumber
+}
+
+// TODO: Description
+export async function redeemDeposits(
```

Comment:
I think we need to modify that function and add integration with the Bridge. A redemption key is built using two things:
- Wallet public key hash
- Redeemer output script

The first one can be derived from the `walletPrivateKey` (taking the public key and then using HASH160 on it). The second one can be derived from the BTC address passed by the redeemer. All other information is held by the Bridge. That means we should:
- Remove the `RedemptionRequest` interface altogether. It will be required in future for requesting redemptions but with more fields so we can safely get rid of it now
- Pass just the `redeemerAddresses` array instead of `redemptionRequests`
- Build redemption requests keys and fetch those redemptions from the Bridge
- Compute the acceptable redemption amount for each redemption to determine the correct output value
- Build and broadcast the transaction

Reply:
I think I've done it in recent commits.
The `redeemDeposits` function has `redeemerAddresses`, `walletPrivateKey` and `bridge` passed as parameters
and builds the key and gets the pending redemption requests from the `Bridge`.

The `createRedemptionTransaction` function does the assembling of the transaction.